Source/Core/InputCommon/ControllerInterface/GameController/GameControllerDevice.mm

```diff
@@ -0,0 +1,109 @@
+// Copyright 2013 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "InputCommon/ControllerInterface/GameController/GameControllerDevice.h"
+
+#include "Common/Logging/Log.h"
+#include "Common/MathUtil.h"
+
+namespace ciface::GameController
+{
+using MathUtil::GRAVITY_ACCELERATION; // 9.80665
+
+Controller::Controller(const GCController* controller) : m_controller(controller)
+{
+  std::vector<std::pair<std::string, GCControllerButtonInput*>> pairs;
+    
+  for (GCControllerButtonInput* item in controller.physicalInputProfile.allButtons)
+  {
+    pairs.push_back(std::make_pair(std::string([item.aliases.allObjects[0] UTF8String]), item));
```

Comment:
```suggestion
    pairs.emplace_back(std::string([item.aliases.allObjects[0] UTF8String], item);
```

Reply:
Is item.aliases.allObjects[0] guaranteed to be valid?

Reply:
According to https://developer.apple.com/documentation/gamecontroller/gccontrollerelement, the aliases object is always present, but I will add a check if the array contains more than 0 entries